### DISCUSSION TITLE
Which maven repository should I use? I always have problems importing dependencies

### DISCUSSION THREAD

**Discussion by sunheyi6** [apache/pulsar#19575] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/50973219/220227743-409e51db-716b-4913-971f-0b22f3b2b2bf.png)

**Comment by tisonkun** [comment, marked as answer]
I use `https://repo1.maven.org/maven2/` which is the default repository and have no problem.

It seems you use aliyun's mirror and you may contact with the vendor.

**Reply by tisonkun** [reply]
From https://maven.aliyun.com/, it seems that `https://maven.aliyun.com/repository/public` is mirror for jcenter, and you may use `https://maven.aliyun.com/repository/central` instead which is mirror for https://repo1.maven.org/maven2/

**Reply by tisonkun** [reply]
What's more, JFrog has shutdown the JCenter repository https://blog.gradle.org/jcenter-shutdown (make it read-only), so new versions of dependencies can be missing. I guess it's the root cause of your case.

**Reply by sunheyi6** [reply]
Either this https://maven.aliyun.com/repository/central or this https://maven.aliyun.com/repository/central  won't work.
![image](https://user-images.githubusercontent.com/50973219/220502590-419e8531-2d91-436f-9e4d-942a67c3a1e4.png)

Can I have a look at your settings file?

**Reply by tisonkun** [reply]
An empty `.m2/settings.xml` can work for me.

You may try to remove the `.m2` folder and reinstall. Or just use `mvn -U` as suggested.

**Reply by sunheyi6** [reply]
ok，dependency can be downloaded successfully.But there is a new problem 
> 'parent.relativePath' of POM org.apache.pulsar:pulsar:2.12.0-SNAPSHOT (D:\java\pulsar\pom.xml) points at com.example:leetcode instead of org.apache:apache, please verify your project structure

**Reply by tisonkun** [reply]
> 'parent.relativePath' of POM org.apache.pulsar:pulsar:2.12.0-SNAPSHOT (D:\java\pulsar\pom.xml) points at com.example:leetcode instead of org.apache:apache, please verify your project structure

This must be about your local env settings. I can do nothing unless know your local env. Please investigate by yourself like googling or ask on SO - it's not an issue of Pulsar.
